core/common/src/main/java/alluxio/underfs/Fingerprint.java

```diff
@@ -73,6 +80,27 @@ public static Fingerprint create(String ufsName, UfsStatus status) {
     if (status == null) {
       return new Fingerprint(Collections.emptyMap());
     }
+    return new Fingerprint(Fingerprint.createInternal(ufsName, status));
+  }
+
+  /**
+   * Parses the input string and returns the fingerprint object.
+   *
+   * @param ufsName the name of the ufs, should be {@link UnderFileSystem#getUnderFSType()}
+   * @param status the {@link UfsStatus} to create the fingerprint from
+   * @param acl the {@link AccessControlList} to create the fingerprint from
+   * @return the fingerprint object
+   */
+  public static Fingerprint create(String ufsName, UfsStatus status, AccessControlList acl) {
+    if (status == null) {
+      return new Fingerprint(Collections.emptyMap());
+    }
+    Map<Tag, String> tagMap = Fingerprint.createInternal(ufsName, status);
+    tagMap.put(Tag.ACL, acl.toString());
+    return new Fingerprint(tagMap);
+  }
+
+  private static Map<Tag, String> createInternal(String ufsName, UfsStatus status) {
```

Comment:
If this is returning a `Map`, maybe this should be called `createTags` or something. `createInternal` sounds like it will create a `Fingerprint`. Also, can you add javadoc for this method?

Reply:
done